File: src/onion/compiler/SymbolTable.java
/* ************************************************************** *
 *                                                                *
 * Copyright (c) 2005, Kota Mizushima, All rights reserved.       *
 *                                                                *
 *                                                                *
 * This software is distributed under the modified BSD License.   *
 * ************************************************************** */
package onion.compiler;

import java.util.HashMap;
import java.util.Map;

public class SymbolTable {
  private Map table;
  
  public SymbolTable() {
    this.table = new HashMap();
  }
  
  public void put(Symbol key, Object value) {
    table.put(key, value);
  }
  
  public Object get(Symbol key) {
    return table.get(key);
  }
  
  public boolean containsKey(Symbol key) {
    return table.containsKey(key);
  }
}

File: src.scala/onion/compiler/SymbolTable.scala
/* ************************************************************** *
 *                                                                *
 * Copyright (c) 2005, Kota Mizushima, All rights reserved.       *
 *                                                                *
 *                                                                *
 * This software is distributed under the modified BSD License.   *
 * ************************************************************** */
package onion.compiler

import collection.mutable.HashMap


class SymbolTable {
  private val table = new HashMap[Symbol, AnyRef]

  def put(key: Symbol, value: AnyRef): Unit =  table.put(key, value)

  def get(key: Symbol): AnyRef = table.get(key)

  def containsKey(key: Symbol): Boolean = table.contains(key)
}